core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -621,12 +626,18 @@ public void reload() {
      *
      * @return true if the version number is not 'DEV'.
      */
+    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
     public boolean isProductionEnvironment() {
         // First is if Blossom runs, second is if Blossom doesn't run
         //noinspection ConstantConditions,MismatchedStringCase - changes in production
         return !("git-local/dev-0000000".equals(GeyserImpl.GIT_VERSION) || "${gitVersion}".equals(GeyserImpl.GIT_VERSION));
     }
 
+    public boolean isDevBuild() {
+        //noinspection ConstantValue - changes in production
+        return !BRANCH.equals("master") && !REPOSITORY.equalsIgnoreCase("https://github.com/GeyserMC/Geyser");
```

Comment:
I personally think that this is better suited in the buildscript which returns a boolean or something that determines whether it is a dev build.

Reply:
https://github.com/GeyserMC/Geyser/pull/4336/commits/1fe027e712cb0f30b7ae8c75191d5f5d18512633
Is this better?